### PR TITLE
[UI Test] - Add simple physical product UI Test

### PR DESCRIPTION
**Pull request by jostnes**
### Description
This PR adds a new test case `Add new product - Simple physical product` to the UI test suite. I've removed the first product currently on the Products list so that that product can be the new product. Also, I realized that the Create Order tests are dependent on this value on the mock file, so I've updated some of the orders mock with the new values. 

To keep the test light while still maintaining the test's purpose (to validate that adding a simple physical product works), only the product name is added from the UI, and other data are added to the mock file. 

The validation is on the new product screen where the test validates the existence of the Save button and an announcement (both of which do not appear when selecting the product from the product list)

### Testing instructions
The new test `test_add_simple_physical_product()` should pass in CI and locally. The test should also fail when verification points are not met.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift [hunk 1]
```diff
@@ -75,4 +75,15 @@ public final class ProductsScreen: ScreenObject {
...
+
+    public func tapAddProduct() throws -> Self {
+        app.buttons["product-add-button"].tap()
+        return self
+    }
+
+    public func selectProductType(productType: String) throws -> SingleProductScreen {
+        app.staticTexts[productType].tap()
+
+        return try SingleProductScreen()
```
Comment (pachlava) [inline WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift]: Just noticing that there's a line break before `return`, while in two functions above there isn't:

```suggestion
        app.staticTexts[productType].tap()
        return try SingleProductScreen()
```

File: WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift [hunk 2]
```diff
@@ -30,4 +30,20 @@ public final class SingleProductScreen: ScreenObject {
 
         return self
     }
+
+    public func addProductTitle(productTitle: String) throws -> Self {
+        app.cells["product-title"].tap()
+        app.cells["product-title"].typeText(productTitle)
```
Comment (pachlava) [inline WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift]: Tap followed by entering text is present in `clearAndEnterText` from `XCTest+Extensions`:

```suggestion
        app.cells["product-title"].clearAndEnterText(text: productTitle)
```

The downside of it is that it also tries to clear existing text before entering, which takes time, so I'd recommend extracting the tap+type lines into a separate `XCUIElement.enterText` and call it instead.

File: WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift [hunk 3]
```diff
@@ -30,4 +30,20 @@ public final class SingleProductScreen: ScreenObject {
...
+    }
+
+    public func publishProduct() throws -> Self {
+        app.buttons["publish-product-button"].tap()
+        return self
+    }
+
+    public func verifyNewProductScreenLoaded() {
+        XCTAssertTrue(app.buttons["save-product-button"].exists)
+        XCTAssertTrue(app.staticTexts["TIP"].exists)
```
Comment (pachlava) [inline WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift:46]: WDYT of checking that this screen contains the product name?

File: WooCommerce/WooCommerceUITests/Tests/ProductsTests.swift [hunk 4]
```diff
@@ -3,6 +3,14 @@ import XCTest
 
 final class ProductsTests: XCTestCase {
 
+    let productTypes = [
+        "physical": "Simple physical product",
+        "virtual": "Virtual physical product",
```
Comment (pachlava) [inline WooCommerce/WooCommerceUITests/Tests/ProductsTests.swift]: I think it should say

```suggestion
        "virtual": "Simple virtual product",
```

SCR-20221212-rfw

**Review by pachlava — COMMENTED** [review]
Hi @jostnes 🙂 👋 

Thanks for the new test! I especially like the parts related to adding new accessibility identifiers 👍.

There's a typo in the products types list for the virtual product, this is the only reason why I'm opting for "Comment"-type review, and not approving.

I also left some other notes - feel free to consider them.

File: WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift (hunk 2 above)
Comment (jostnes) [reply WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift]: the initial version of this was using `clearAndEnterText()` but clearing the text field isn't needed in this case so i changed it. good idea on extracting only tap and type, i'll do that! thanks for the suggestion!

File: WooCommerce/WooCommerceUITests/Tests/ProductsTests.swift (hunk 4 above)
Comment (jostnes) [reply WooCommerce/WooCommerceUITests/Tests/ProductsTests.swift]: oof - such a bad typo, thanks for catching this! will fix it

Comment (jostnes) [reply WooCommerce/WooCommerceUITests/Tests/ProductsTests.swift]: updated in https://github.com/woocommerce/woocommerce-ios/pull/8380/commits/91b15ef1f55c562f15434b7d1f7e8f3459308652#diff-ba22d4d1543946cc905fc3a52bdd2c2533cb2475eb78c9fb1e9bef76c9870fb9

File: WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift (hunk 3 above)
Comment (jostnes) [reply WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift:46]: updated in https://github.com/woocommerce/woocommerce-ios/pull/8380/commits/91b15ef1f55c562f15434b7d1f7e8f3459308652#diff-38e7198d7b53c64cf1257404887931cc42ac178877c05d5bd98a463ac1da1d0b

File: WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift (hunk 2 above)
Comment (jostnes) [reply WooCommerce/UITestsFoundation/Screens/Products/SingleProductScreen.swift]: updated in https://github.com/woocommerce/woocommerce-ios/pull/8380/commits/91b15ef1f55c562f15434b7d1f7e8f3459308652

**Review by pachlava — APPROVED** [review]
Thanks for all the follow-ups and changes @jostnes! 🎉 

I took the liberty of making a tiny commit 718db716f10f39178af0385e5d95f7a3fe7c1522, to save up on back and forth's, so that you can merge it once you're back online.

It now looks a bit questionable to approve a PR in which I made a commit, but I see no reasons why it can't be approved 😅.

Thank you once again! :shipit:

**Comment by jostnes** [comment]
I think it's fine, since the change you made is a small change that I missed 😅  Thank you for the thorough review as always! 🚀

**Pull request merged**